pkg/kubectl/cmd/set/set_env.go

```diff
@@ -231,65 +307,40 @@ func (o *EnvOptions) RunEnv(f cmdutil.Factory) error {
 		return err
 	}
 
-	if len(o.From) != 0 {
-		b := f.NewBuilder().
-			Internal().
-			LocalParam(o.Local).
-			ContinueOnError().
-			NamespaceParam(cmdNamespace).DefaultNamespace().
-			FilenameParam(enforceNamespace, &o.FilenameOptions).
-			Flatten()
-
-		if !o.Local {
-			b = b.
-				LabelSelectorParam(o.Selector).
-				ResourceTypeOrNameArgs(o.All, o.From).
-				Latest()
-		}
-
-		infos, err := b.Do().Infos()
-		if err != nil {
-			return err
+	for _, s := range secrets {
+		for key := range s.Data {
+			envVar := v1.EnvVar{
```

Comment:
This logic looks the same in the configmaps loop, this inner loop could be placed in a separate function that returns []v1.EnvVar

Reply:
Sorry, typo by me, the struct contents should be different for both resources